File: android/app/src/main/java/com/bridgingproject/MyNativeModule.java
package com.bridgingproject;

import android.widget.Toast;

import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

public class MyNativeModule extends ReactContextBaseJavaModule {

    public MyNativeModule(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public String getName() {
        return "MyNativeModule";
    }

    @ReactMethod
    public void callNativeAPI(String params, Promise promise) {
        try {
            String result = "Native API called>>> " + params;
            promise.resolve(result);
        } catch (Exception e) {
            promise.reject("Error ", e);
        }
    }
}

File: ios/MyNativeModule.swift
import Foundation

@objc(MyNativeModule)
class MyNativeModule: NSObject {

    @objc(callNativeAPI:resolver:rejecter:)
    func callNativeAPI(params: String, resolver: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock) {
        let result = "Native API called>>> \(params)"
        resolver(result)
    }
}

